pyccel/parser/syntactic.py

```diff
@@ -631,6 +631,7 @@ def _visit_FunctionDef(self, stmt):
         global_vars  = []
         headers      = []
         templates    = {}
+        template    = {}
```

Comment:
`templates` is a temporarily variable, maybe it's better if it's mentioned in the name, I would use `tmp_templates` and `templates`

Reply:
i think its not, because its a dict that is manipulated any a whole if statement.